apps/google-analytics-4/frontend/src/components/main-app/analytics-app/AnalyticsApp.tsx

```diff
@@ -14,24 +14,24 @@ const DEFAULT_ERR_MSG = 'Oops! Cannot display the analytics data at this time.';
 const EMPTY_DATA_MSG = 'There are no page views to show for this range';
 
 interface Props {
-  serviceAccountKeyId: ServiceAccountKeyId;
-  serviceAccountKey: ServiceAccountKey;
+  api: Api;
   propertyId: string;
-  reportSlug: string;
+  slugFieldInfo: ContentTypeValue;
 }
 const AnalyticsApp = (props: Props) => {
-  const { serviceAccountKeyId, serviceAccountKey, propertyId, reportSlug } = props;
+  const { api, propertyId, slugFieldInfo } = props;
+  const { slugField, urlPrefix } = slugFieldInfo;
   const [runReportResponse, setRunReportResponse] = useState<RunReportData>({} as RunReportData);
   const [dateRange, setDateRange] = useState<DateRangeType>('lastWeek');
   const [startEndDates, setStartEndDates] = useState<any>(getRangeDates('lastWeek')); // TYPE
-  const [slugValue] = useFieldValue<string>('slug');
+  const [slugValue] = useFieldValue<string>(slugField);
   const [loading, setLoading] = useState<boolean>(true);
   const [error, setError] = useState<Error>();
 
-  const api = useApi(serviceAccountKeyId, serviceAccountKey);
-
   useAutoResizer();
 
+  const reportSlug = `${urlPrefix || ''}${slugValue || ''}`;
```

Comment:
The legacy code sends an array of these slugs in the form of 'filters', to account for a trailing '/' and a non-trailing '/'. You can see the code [here](https://github.com/contentful/apps/blob/85a95f9c800caedcbc20c7a99ce115b9c742fd48/apps/google-analytics/src/Timeline.tsx#L102). I am not sure why this is the case but feel they must have added it for a reason ?. Right now the accepted `slug` type is only a string at the root of the `googleApiService`. Since it is a larger change I haven't implemented it until validating that the methodology behind these slugs and page paths is correct, and that this formatting is warranted. Keeping an eye on it 👍

Reply:
Yeah! so I our URL is:

prefix + slug

and both are user input. (one in the app config screen and the other on the entry itself).

for prefix especially people might put:

* `about/blog`
* `about/blog/`
* `/about/blog/`
* `/about/blog`

we either need to validate those at entry or normalize them in your request.

and i'm not sure how google treats this either 🤔 

if you want to punt on this to a follow up PR, maybe drop a comment here and add a subtask?

Reply:
https://contentful.atlassian.net/browse/INTEG-256